azurelinuxagent/common/logcollector.py

```diff
@@ -19,59 +19,83 @@
 
 import glob
 from heapq import heappush, heappop
+import logging
 import os
 import subprocess
 import zipfile
 
+from azurelinuxagent.common.conf import get_lib_dir, get_ext_log_dir
 from azurelinuxagent.common.future import ustr
-from azurelinuxagent.common.utils.fileutil import mkdir, read_file, rm_files, append_file, append_items_to_file
+from azurelinuxagent.common.utils.fileutil import mkdir, read_file, write_file, rm_files
 
+_EXTENSION_LOG_DIR = get_ext_log_dir()
+_AGENT_LIB_DIR = get_lib_dir()
 
-_LOG_COLLECTOR_DIR = '/var/lib/waagent/logcollector'
-_TRUNCATED_FILES_DIR = '/var/truncated'
+_LOG_COLLECTOR_DIR = os.path.join(_AGENT_LIB_DIR, "logcollector")
+_TRUNCATED_FILES_DIR = "/var/truncated"
 
 _OUTPUT_RESULTS_FILE_PATH = os.path.join(_LOG_COLLECTOR_DIR, "results.txt")
-COMPRESSED_ARCHIVE_PATH = os.path.join(_LOG_COLLECTOR_DIR, "logs.zip")
+_COMPRESSED_ARCHIVE_PATH = os.path.join(_LOG_COLLECTOR_DIR, "logs.zip")
 
 _MUST_COLLECT_FILES = [
-    '/var/log/waagent.log',
-    '/var/lib/waagent/GoalState.*.xml',
-    '/var/lib/waagent/ExtensionsConfig.*.xml',
-    '/var/lib/waagent/waagent_status.json',
-    '/var/lib/waagent/history/*.zip',
-    '/var/log/azure/*/*',
-    '/var/log/azure/*/*/*',
-    '/var/lib/waagent/HostingEnvironmentConfig.xml',
-    '/var/log/waagent*',
+    "/var/log/waagent.log",
```

Comment:
shouldn't hardcode this path

Reply:
This is hardcoded throughout the agent. Unlike `/var/lib/waagent` and `/var/log/azure`, we don't expose a parameter for agent log file.

Reply:
I remember this was part of the log-rotate PR that Varad was working on. It might be useful for you to parameterize it now and reuse the same one for the other PR (or do it the other way round), whatever works for you I guess.

Reply:
> This is hardcoded throughout the agent. 

It shouldn't. Please add a function similar to get_ext_log_dir and let's start using it in new code

Reply:
`get_ext_log_dir` reads from the config file, and currently we don't have an entry in the config file for agent log file. I wouldn't want to do that change in the scope of this PR. What I can do is parameterize it for the log collector (add `$AGENT_LOG` to the manifests) and then define the the value in this module as `/var/log/waagent.log` until we make the change throughout the agent, including the config file.